**Epigenomics tools**
=====================

Below are various tools used for Epigenomics sequencing data analysis

`MACS (Model-based Analysis of ChIP-Seq) <https://github.com/macs3-project/MACS>`_: A well-established tool for peak calling in ChIP-Seq data, helping researchers identify regions of protein-DNA binding such as histone modifications.

`SICER2 (Spatial Clustering for Identification of ChIP-Enriched Regions) <https://zanglab.github.io/SICER2/>`_: This tool is designed for ChIP-Seq data to identify broader regions of enrichment, especially useful for studying histone modifications.

`DSS (Dispersion Shrinkage for Sequencing) <https://www.bioconductor.org/packages/devel/bioc/vignettes/DSS/inst/doc/DSS.html>`_: An R package that enables differential methylation analysis, useful for comparing methylation levels between conditions.

`CHIPseeker <https://www.bioconductor.org/packages/devel/bioc/vignettes/ChIPseeker/inst/doc/ChIPseeker.html>`_: Primarily used for annotating and visualizing ChIP-seq peaks, CHIPseeker helps determine the genomic regions where ChIP-seq peaks are located, such as promoters, exons, or intergenic regions.

`deepTools <https://deeptools.readthedocs.io/en/develop/>`_: deepTools is a versatile suite designed for visualizing and analyzing various types of high-throughput sequencing data, including ChIP-seq, RNA-seq, and ATAC-seq.

`SEACR (Sparse Enrichment Analysis for CUT&RUN) <https://github.com/FredHutch/SEACR>`_: SEACR is a peak caller specifically designed for CUT&RUN (Cleavage Under Targets & Release Using Nuclease) data, which is an alternative to traditional ChIP-seq that requires fewer cells and provides higher signal-to-noise ratios.
