**Variant analysis tools**
==========================

Under this category are all tools that can be utilized for variant analysis.

**Variant Calling**

`GATK <https://gatk.broadinstitute.org/hc/en-us/categories/360002302312-Getting-Started>`_ (Genome Analysis Toolkit): One of the most comprehensive tools for variant calling (SNPs, indels), with a suite of tools for pre-processing, calling, and post-processing variants.

`FreeBayes <https://bioinformaticsworkbook.org/dataAnalysis/VariantCalling/freebayes-dnaseq-workflow.html#gsc.tab=0>`_: For detecting SNPs, indels, and other small polymorphisms in aligned reads.

`Samtools <https://github.com/arq5x/tutorials/blob/master/samtools.md>`_ (mpileup): A widely used variant caller in the SAMtools suite for calling SNPs and indels.

`DeepVariant <https://github.com/google/deepvariant/blob/r1.6.1/docs/deepvariant-details.md>`_: A Google-developed deep learning-based tool for variant calling that provides highly accurate calls.

`Platypus <https://rahmanteamdevelopment.github.io/Platypus/documentation.html>`_: A tool for SNP and indel variant calling designed for analyzing small cohorts.

**Copy Number Variation (CNV) Detection**

`CNVnator <https://github.com/abyzovlab/CNVnator>`_: For detecting CNVs using read depth information from aligned sequences.

`Control-FREEC <https://boevalab.inf.ethz.ch/FREEC/tutorial.html>`_: For detection of copy number and allelic imbalances, often used in tumor samples.

`ExomeDepth <https://rdrr.io/cran/ExomeDepth/f/inst/doc/ExomeDepth-vignette.pdf>`_: Specifically designed for detecting CNVs in whole-exome sequencing data.

`Manta <https://github.com/Illumina/manta/blob/master/docs/userGuide/README.md>`_: A tool for calling structural variants and indels in addition to copy number variants.

**Variant Annotation**

`ANNOVAR <https://annovar.openbioinformatics.org/en/latest/user-guide/startup/>`_: One of the most popular tools for functional annotation of genetic variants, providing information on gene functions and variant effects.

`SnpEff <https://pcingola.github.io/SnpEff/>`_: For variant annotation and effect prediction, particularly efficient in large datasets.

`SnpEff <https://useast.ensembl.org/info/docs/tools/vep/script/vep_tutorial.html>`_ (Variant Effect Predictor): Developed by Ensembl, annotates variants and predicts their effects on genes and transcripts.

`ClinVar <https://guides.nnlm.gov/tutorial/ncbi-clinvar>`_: Used as a database and annotation tool to interpret clinical significance in variants.

