**RNA-seq analysis tools**
==========================

Below are the list of web applications that can be used to carry out RNA-seq analysis

.. list-table:: RNA-seq analysis tools
   :widths: 15 65 20
   :header-rows: 1

   * - Resource
     - Access
     - Description
   * - RaNA-seq
     - RaNA-Seq is an open bioinformatics tool for the quick analysis of RNA-Seq data. It performs a full analysis in minutes quantifying FASTQ files, calculating quality control metrics, running differential expression analyses and enabling the interpretation of results with functional analyses. Their analysis pipeline integrates cutting-edge bioinformatics tools and simplifies its application with a friendly Web interface designed for non-experienced users in these analyses. Each analysis can be customized setting up input parameters and applies generally accepted and reproducible protocols.
     - `Link to resource <https://ranaseq.eu/home>`_
   * - BEAVR
     - BEAVR is an easy-to-use tool that facilitates interactive analysis and exploration of RNA-seq data. BEAVR is developed in R and uses DESeq2 as its engine for differential gene expression (DGE) analysis, but assumes users have no prior knowledge of R or DESeq2. BEAVR allows researchers to easily obtain a table of differentially-expressed genes with statistical testing and then visualize the results in a series of graphs, plots and heatmaps.
     - `Link to resource <https://github.com/developerpiru/BEAVR>`_
   * - iDEP
     - iDEP is a bioinformatics platform for analyzing gene expression data graphically, interactively, and reproducibly. The input file is a gene-level expression matrix derived from RNA-Seq, microarray, proteomics, or other methods. Hosted at South Dakota State Univerisity, iDEP is developed as an R package based on the Golem framework. It seamlessly connects 63 R/Bioconductor packages, 2 web services, and comprehensive annotation and pathway databases for 220 plant and animal species. The workflow can be reproduced by downloading customized R code and related pathway files.
     - `Link to resource <http://ge-lab.org/idep/>`_
   * - STARTapp
     - The START App is a web-based application written entirely in the open-source R programming language (R Core Team, 2015) using the Shiny framework. It is fully cross-platform and can be launched locally from any computer with R installed. Alternatively, users can host their own version of the app with their transcriptome data on a local or remote server so that other users may access the app and data from a website without installing R or the application.
     - `Link to resource <https://kcvi.shinyapps.io/START/>`_
   * - Degust
     - Degust is a web-based tool designed to facilitate the exploration, analysis, and visualization of RNA-Seq data, particularly focusing on differential gene expression (DGE) analysis. Developed by David R. Powell at Monash University, Degust offers an interactive platform that simplifies the interpretation of complex RNA-Seq datasets. 
     - `Link to resource <https://degust.erc.monash.edu/>`_
