**RNA-seq analysis tools**
=================

Below are all tools that are frequently used to carry out RNA-seq analysis.

`DESeq2 <https://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html>`_: For differential expression analysis, particularly in RNA-seq datasets.

`edgeR <https://www.bioconductor.org/packages/devel/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf>`_: Another tool for differential expression analysis, particularly suited for RNA-seq experiments with small sample sizes.

`Cufflinks <http://homer.ucsd.edu/homer/basicTutorial/rnaseqCufflinks.html>`_: For transcript assembly, quantification, and differential expression analysis.

`STAR <https://physiology.med.cornell.edu/faculty/skrabanek/lab/angsd/lecture_notes/STARmanual.pdf>`_ (Spliced Transcripts Alignment to a Reference): A fast aligner for RNA-seq data, especially for spliced reads.

`TopHat <https://ccb.jhu.edu/software/tophat/manual.shtml>`_: Splice-aware aligner for RNA-seq reads (often used in tandem with Cufflinks).
`RSEM <https://github.com/bli25/RSEM_tutorial/blob/master/README.md>`_ (RNA-Seq by Expectation-Maximization): Tool for accurate transcript quantification from RNA-seq data.

`Kallisto <https://pachterlab.github.io/kallisto/manual>`_: Provides fast and accurate transcript quantification using pseudo-alignment.

`Sleuth <https://pachterlab.github.io/sleuth/manual>`_: Works with Kallisto output to perform differential expression analysis. 
